File: 2024/solutions/day01/go/main_test.go
package main

import (
	"log"
	"os"
	"testing"
)

func getExample() string {
	data, err := os.ReadFile("../example.txt")
	if err != nil {
		log.Fatalf("Failed to read file %v", err)
	}

	return string(data)
}

func getInput() string {
	data, err := os.ReadFile("../input.txt")
	if err != nil {
		log.Fatalf("Failed to read file %v", err)
	}

	return string(data)
}

func TestPartOne(t *testing.T) {
	example := getExample()
	input := getInput()
	t.Run("example", func(t *testing.T) {
		actual := partOne(example)
		expected := 11
		if actual != expected {
			t.Errorf("\nactual: %v\nexpected: %v", actual, expected)
		}
	})

	t.Run("input", func(t *testing.T) {
		actual := partOne(input)
		expected := 2285373
		if actual != expected {
			t.Errorf("\nactual: %v\nexpected: %v", actual, expected)
		}
	})
}

func TestPartTwo(t *testing.T) {
	example := getExample()
	input := getInput()
	t.Run("example", func(t *testing.T) {
		actual := partTwo(example)
		expected := 31
		if actual != expected {
			t.Errorf("\nactual: %v\nexpected: %v", actual, expected)
		}
	})

	t.Run("input", func(t *testing.T) {
		actual := partTwo(input)
		expected := 21142653
		if actual != expected {
			t.Errorf("\nactual: %v\nexpected: %v", actual, expected)
		}
	})
}

File: 2024/solutions/day09/typescript/main.test.ts
import { expect, test, describe } from "bun:test";
import input from "../input.txt";
import example from "../example.txt";
import { partOne, partTwo } from "./main.ts";

describe("Day 09", () => {
  test("part1, example", () => {
    const actual = partOne(example);
    const expected = 1928;
    console.log(actual);
    expect(actual).toBe(expected);
  });

  test("part1, input", () => {
    const actual = partOne(input);
    const expected = 6154342787400;
    console.log(actual);
    expect(actual).toBe(expected);
  });
  //
  test("part2, example", () => {
    const actual = partTwo(example);
    const expected = 2858;
    console.log(actual);
    expect(actual).toBe(expected);
  });

  test("part2, input", () => {
    const actual = partTwo(input);
    const expected = 6183632723350;
    console.log(actual);
    expect(actual).toBe(expected);
  });
});
